### DISCUSSION TITLE
Quick terminal windows don't display over fullscreen apps

### DISCUSSION THREAD

**Discussion by tanguy-michardiere** [ghostty-org/ghostty#3719] [Feature Requests, Ideas] [answered]
When I'm on a fullscreen app and use my configured keybind for opening a quick terminal window, the fullscreen app loses focus but no terminal window is displayed.

If I type some text in this state I cannot see it, but if I open the quick terminal window later without a fullscreen app it shows the text I typed. So in a way, the quick terminal window is opened and properly focused but "behind" the fullscreen app.

I tested this on several apps, some native and some third-party.

I am using MacOS Sequoia 15.2 and Ghostty 1.0.0 from homebrew:
```
Ghostty 1.0.0

Version
  - version: 1.0.0
  - channel: stable
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.none
  - font engine: font.main.Backend.coretext
  - renderer   : renderer.Metal
  - libxev     : main.Backend.kqueue
 ```

My complete configuration:
```
theme = Dark+
font-family = FiraMono Nerd Font Mono

quick-terminal-animation-duration = 0

auto-update = off


keybind = global:super+grave_accent=toggle_quick_terminal
```

**Comment by mitchellh** [comment]
What is the behavior of iTerm2 in this situation? Does it show up over the existing fullscreen window? I believe it does but just want to check. I can check myself later but figured it's easy to ask. 😄

**Reply by ericmarcos** [reply]
Yes, iTerm behaves as expected (it displays the terminal over the fullscreen window) if you configure a Hotkey Window with the "Floating Window" setting enabled:

<img width="585" alt="Screenshot 2025-01-01 at 23 46 07" src="https://github.com/user-attachments/assets/71f1deb7-c2c8-4d53-b51c-e70846e29730" />

**Comment by mitchellh** [comment, marked as answer]
This is fixed on tip and will be part of Ghostty 1.1
